### PR TITLE
Renaming "runs" to "steps"

### PR DESCRIPTION
**Pull request by taylordowns2000**
### Readme

1. This is only part of #1571 but can be deployed as is.
2. It renames `AttemptRuns` to `AttemptSteps`.
3. It renames `Runs` to `Steps`.
4. It does _not_ rename `Attempts` to `Runs`, but now the two words **_can be used interchangeably_** because each of those words refers to the **_same referent_**.
    - It's not my intention to keep both terms around much longer, but getting this first part of the renaming done will reduce the chance of communication errors significantly. 
   - Consider: if someone says "run" _or_ "attempt", they can only be referring to the thing which the worker will take and execute for a given work order. Whether you call it a run or an attempt, it's directly related to a work order and it can have one or many steps.
   - The temporary hierarchy looks like: `WorkOrder > Run/Attempt > Step`
5. Take a look at how it handles conversion of `run:start` to `step:start` in the `AttemptChannel` and shows a warning.
6. This also removes a whole bunch of unused code, including the oban `runs` queue and related telemetry metrics.

### After this PR is merged, what's next? 
1. Next, in another PR we need to rename indexes/constraints/foreign_keys and determining how to handle the `log_lines` partitions: #1655
2. The JSON api for runs (which is important for GovStack compliance) needs to be updated but it should point to what we now call `Attempts`, not to `steps`: #1656

## Review checklist

- [x] I have performed a **self-review** of my code
- [x] I have verified that all appropriate **authorization policies** have been implemented and tested
- [x] If needed, I have updated the **changelog**
- [x] Product has **QA'd** this feature

### CODE REVIEW

**Review by stuartc — COMMENTED** [review]
This is great @taylordowns2000, well done! :raised_hands: 

- We need a plan for deployment, order of operations and tolerance for downtime 
- Additions to the migrations (at least the fk constraint names in order to match up with the ecto default constraint names)
- Whats happening to the commented out `*/runs` JSON api?

File: lib/lightning_web/channels/attempt_channel.ex [hunk 1]
```diff
@@ -230,4 +240,16 @@ defmodule LightningWeb.AttemptChannel do
     :ok = Scrubber.add_samples(scrubber, samples, basic_auth)
     {:ok, scrubber}
   end
+
+  defp worker_upgrade_required(v),
+    do:
+      Logger.warning("Please upgrade your connect ws-worker to #{v} or greater")
```
[the remark this answers is not in the document]
Comment (taylordowns2000) [reply lib/lightning_web/channels/attempt_channel.ex:246]: I don't expect a ws-worker that handles this for ___ days/weeks. What's the downside of doing this in the meantime?

File: lib/lightning_web/router.ex [hunk 2]
```diff
@@ -54,11 +54,11 @@ defmodule LightningWeb.Router do
 
     resources "/projects", API.ProjectController, only: [:index, :show] do
       resources "/jobs", API.JobController, only: [:index, :show]
-      resources "/runs", API.RunController, only: [:index, :show]
+      # resources "/runs", API.RunController, only: [:index, :show]
     end
 
     resources "/jobs", API.JobController, only: [:index, :show]
-    resources "/runs", API.RunController, only: [:index, :show]
+    # resources "/runs", API.RunController, only: [:index, :show]
```
[the remark this answers is not in the document]
Comment (taylordowns2000) [reply lib/lightning_web/router.ex:61]: See next issue!

File: lib/lightning_web/channels/attempt_channel.ex [hunk 3]
```diff
@@ -230,4 +240,16 @@ defmodule LightningWeb.AttemptChannel do
...
+
+  # TODO - remove this once the migration is complete
+  defp rename_run_id(%{"run_id" => id} = map) do
+    Map.delete(map, "run_id")
+    |> Map.put("step_id", id)
+  end
+
+  defp rename_run_id(any), do: any
```
[the remark this answers is not in the document]
Comment (stuartc) [reply lib/lightning_web/channels/attempt_channel.ex:257]: To be improved down the line, _after_ https://github.com/OpenFn/kit/issues/563 lands.

[the remark this answers is not in the document]
Comment (taylordowns2000) [reply lib/lightning_web/channels/attempt_channel.ex:257]: added todos. thanks for link to kit/563

**Pull request merged**